File: frontend/src/services/favoritesService.ts
import api from './api';
import type { Vendor } from '@/types';

export interface Favorite {
  id: string;
  vendorId: string;
  createdAt: string;
  vendor: Vendor;
}

export async function getFavorites(): Promise<Favorite[]> {
  const { data } = await api.get('/favorites');
  return data;
}

export async function addFavorite(vendorId: string): Promise<Favorite> {
  const { data } = await api.post('/favorites', { vendorId });
  return data;
}

export async function removeFavorite(vendorId: string): Promise<void> {
  await api.delete(`/favorites/${vendorId}`);
}

export async function isFavorite(vendorId: string): Promise<boolean> {
  const { data } = await api.get(`/favorites/check/${vendorId}`);
  return data;
}

File: flutter_geobite/lib/services/favorites_service.dart
import '../core/api_client.dart';

class FavoritesService {
  /// Returns a list of favorite vendor records for the current user.
  Future<List<Map<String, dynamic>>> getFavorites() async {
    try {
      final response = await apiClient.dio.get('/favorites');
      final data = response.data;
      final List list = data is Map ? (data['data'] ?? data['favorites'] ?? []) : data;
      return list.cast<Map<String, dynamic>>();
    } catch (e) {
      throw Exception('Failed to load favorites: $e');
    }
  }

  /// Adds a vendor to the current user's favorites.
  Future<Map<String, dynamic>> addFavorite(String vendorId) async {
    try {
      final response = await apiClient.dio.post('/favorites', data: {'vendorId': vendorId});
      return Map<String, dynamic>.from(response.data);
    } catch (e) {
      throw Exception('Failed to add favorite: $e');
    }
  }

  /// Removes a vendor from the current user's favorites.
  Future<void> removeFavorite(String vendorId) async {
    try {
      await apiClient.dio.delete('/favorites/$vendorId');
    } catch (e) {
      throw Exception('Failed to remove favorite: $e');
    }
  }
}

final favoritesService = FavoritesService();
